app/code/Magento/Catalog/Model/Product.php

```diff
@@ -835,10 +835,10 @@ public function getStoreIds()
                 if ($this->_storeManager->isSingleStoreMode()) {
                     $websiteIds = array_keys($websiteIds);
                 }
-                foreach ($websiteIds as $websiteId) {
-                    $websiteStores = $this->_storeManager->getWebsite($websiteId)->getStoreIds();
-                    $storeIds = array_merge($storeIds, $websiteStores);
-                }
+                $websiteStoreIds = array_map(function (int $websiteId): array {
+                    return $this->_storeManager->getWebsite($websiteId)->getStoreIds();
+                }, $websiteIds);
+                $storeIds = array_merge($storeIds, ...$websiteStoreIds);
```

Comment:
This change is only because the static tests complained about the `array_merge` in a loop. I didn't write that loop, but I don't mind changing it to get the tests green.